jquery.flot.js

```diff
@@ -2224,21 +2221,21 @@ Licensed under the MIT license.
                     case 'none':
                         break;
                     case 'endpoints':
-                        firstLabelBox = drawAxisLabel(axis.ticks[0], []);
-                        lastLabelBox = drawAxisLabel(axis.ticks[axis.ticks.length - 1], [firstLabelBox]);
+                        labelBoxes.push(drawAxisLabel(axis.ticks[0], labelBoxes));
+                        labelBoxes.push(drawAxisLabel(axis.ticks[axis.ticks.length - 1], labelBoxes));
                         break;
                     case 'major':
-                        firstLabelBox = drawAxisLabel(axis.ticks[0], []);
-                        lastLabelBox = drawAxisLabel(axis.ticks[axis.ticks.length - 1], [firstLabelBox]);
+                        labelBoxes.push(drawAxisLabel(axis.ticks[0], labelBoxes));
+                        labelBoxes.push(drawAxisLabel(axis.ticks[axis.ticks.length - 1], labelBoxes));
                         for (i = 1; i < axis.ticks.length - 1; ++i) {
-                            previousLabelBox = drawAxisLabel(axis.ticks[i], [firstLabelBox, previousLabelBox, lastLabelBox]);
+                            labelBoxes.push(drawAxisLabel(axis.ticks[i], labelBoxes));
                         }
                         break;
                     case 'all':
-                        firstLabelBox = drawAxisLabel(axis.ticks[0], []);
-                        lastLabelBox = drawAxisLabel(axis.ticks[axis.ticks.length - 1], [firstLabelBox]);
+                        labelBoxes.push(drawAxisLabel(axis.ticks[0], []));
+                        labelBoxes.push(drawAxisLabel(axis.ticks[axis.ticks.length - 1], labelBoxes));
                         for (i = 1; i < axis.ticks.length - 1; ++i) {
-                            previousLabelBox = drawAxisLabel(axis.ticks[i], [firstLabelBox, previousLabelBox, lastLabelBox]);
+                            labelBoxes.push(drawAxisLabel(axis.ticks[i], labelBoxes));
```

Comment:
The firstLabelBox, previouseLabelBox and lastLabelBox was an 'optimization' to prevent useless comparisons. Apparently, sometimes the width of the tick label is so large that comparing the box of the current label with the previous is not enough - more than 2 boxes can overlap in the same point :)